### DISCUSSION TITLE
Stream to an arrow file in Node.js using `db.arrowIPCStream`

### DISCUSSION THREAD

**Discussion by mhkeller** [duckdb/duckdb#7008] [Q&A]
Is there an example of how to use [`db.arrowIPCStream`](https://duckdb.org/docs/api/nodejs/reference#module_duckdb..Connection+arrowIPCStream) to stream results to an arrow file?

**Comment by mhkeller** [comment]
I've been looking at using the`RecordBatchStreamWriter.writeAll` export from `apache-arrow` like this:

```js
const stream = fs.createWriteStream('test.arrow');

const ipc = await db.arrowIPCStream('select * from iris');
const data = (await RecordBatchStreamWriter.writeAll(ipc))
  .pipe(stream);
```

but the resulting file shows up as empty when I try to read it with this:

```js
const arrow = readFileSync('test.arrow');
const table = tableFromIPC(arrow);

console.table(table.toArray());
```

I've also tried iterating using [this](https://github.com/apache/arrow/blob/main/js/bin/json-to-arrow.js) as an example:

```js
import { promisify } from 'util';
import { finished } from 'stream';

const eos = promisify(finished);

const ipc = await db.arrowIPCStream('select * from iris');

for await (const batch of ipc) {
  const data = RecordBatchReader.from(batch)
    .pipe(RecordBatchStreamWriter.throughNode())
    .pipe(stream);

  await eos(data);
}
```

but no luck.

**Comment by Tishj** [comment]
I'm not that familiar with the node client, or the arrowIPCStream, but after a quick look at our existing tests I found
`extension/arrow_clone/test/nodejs/arrow_ipc.test.js`

That might help as an example

**Reply by Mause** [reply]
That file is here btw:  https://github.com/duckdblabs/arrow/blob/main/test/nodejs/arrow_ipc.test.js

**Reply by mhkeller** [reply]
Thanks for the reference. I'm still not clear on how to stream to an arrow file. I see this, which may be helpful but I haven't got a working pipeline yet: https://github.com/apache/arrow/blob/main/js/test/unit/ipc/writer/streams-node-tests.ts

**Comment by matt-allan** [comment]
This looks like it's currently broken: https://github.com/duckdblabs/arrow/issues/4

**Reply by mhkeller** [reply]
This seems to fixed. Any idea on how to now write to an arrow file?
